packages/mobile/src/tokens/saga.ts

```diff
@@ -266,44 +270,69 @@ export async function getERC20TokenContract(tokenAddress: string) {
   return new kit.web3.eth.Contract(erc20.abi, tokenAddress)
 }
 
-export async function getERC20TokenBalance(token: StoredTokenBalance, address: string) {
-  let balance = null
-  try {
-    const contract = await getERC20TokenContract(token.address)
-    balance = await contract.methods.balanceOf(address).call()
-  } catch (error) {
-    Logger.error(TAG, `error fetching balance for ${token.name}`, error)
-  }
-  return balance
+interface FetchedTokenBalance {
+  tokenAddress: string
+  balance: string
 }
 
-export function* fetchReadableTokenBalance(address: string, token: StoredTokenBalance) {
-  const balance: number | null = yield call(getERC20TokenBalance, token, address)
-  return {
-    ...token,
-    balance:
-      balance || balance === 0
-        ? new BigNumber(balance)
-            .dividedBy(new BigNumber(10).exponentiatedBy(token.decimals))
-            .toString()
-        : null,
+interface UserBalancesResponse {
+  userBalances: {
+    balances: FetchedTokenBalance[]
   }
 }
 
+export async function fetchTokenBalancesForAddress(
+  address: string
+): Promise<FetchedTokenBalance[]> {
+  const response = await apolloClient.query<UserBalancesResponse, { address: string }>({
+    query: gql`
+      query FetchUserBalances($address: Address!) {
+        userBalances(address: $address) {
+          balances {
+            tokenAddress
+            balance
+          }
+        }
+      }
+    `,
+    variables: {
+      address,
+    },
+    fetchPolicy: 'network-only',
+    errorPolicy: 'all',
+  })
+
+  return response.data.userBalances.balances
+}
+
 export function* importTokenInfo() {
```

Comment:
Should we rename this?
```suggestion
export function* fetchTokenBalances() {
```